ci-operator/step-registry/cucushift/upgrade/toversion/cucushift-upgrade-toversion-commands.sh

```diff
@@ -0,0 +1,487 @@
+        if [[ "${TARGET_VERSION}" =~ "-rc" ]]; then
+            expected_target="${TARGET_VERSION%-*}"
+            for version in ${available_versions[*]}; do
+                if [[ "${version}" == "${expected_target}" ]]; then
+                    TARGET_VERSION="${expected_target}"
+                    break
+                fi
+            done
+        fi
+        # latest: run upgrade with "oc adm upgrade --to-latest";
+        if [[ "${UPGRAGE_TO_VERSION}" == "latest" ]]; then
+            upgrade_cmd="oc adm upgrade --to-latest"
+        fi
+    fi
+    TARGET_MINOR_VERSION="$(echo "${TARGET_VERSION}" | cut -f2 -d.)"
+    export TARGET_MINOR_VERSION
+    export TARGET_VERSION
+    echo "Target release version is: ${TARGET_VERSION}\n"
+}
+
+# Monitor the upgrade status
+function check_upgrade_status() {
+    local wait_upgrade="${TIMEOUT}" out avail progress
+    echo "Starting the upgrade checking on $(date "+%F %T")"
+    while (( wait_upgrade > 0 )); do
+        sleep 5m
+        wait_upgrade=$(( wait_upgrade - 5 ))
+        if ! ( run_command "oc get clusterversion" ); then
+            continue
+        fi
+        if ! out="$(oc get clusterversion --no-headers)"; then continue; fi
+        avail="$(echo "${out}" | awk '{print $3}')"
+        progress="$(echo "${out}" | awk '{print $4}')"
+        if [[ ${avail} == "True" && ${progress} == "False" && ${out} == *"Cluster version is ${TARGET_VERSION}" ]]; then
+            echo -e "Upgrade succeed on $(date "+%F %T")\n\n"
+            return 0
+        fi
+    done
+    if [[ ${wait_upgrade} -le 0 ]]; then
+        echo -e "Upgrade timeout on $(date "+%F %T"), exiting\n" && return 1
+    fi
+}
+
+# Check version, state in history
+function check_history() {
+    local version state
+    version=$(oc get clusterversion/version -o jsonpath='{.status.history[0].version}')
+    state=$(oc get clusterversion/version -o jsonpath='{.status.history[0].state}')
+    if [[ ${version} == "${TARGET_VERSION}" && ${state} == "Completed" ]]; then
+        echo "History check PASSED, cluster is now upgraded to ${TARGET_VERSION}" && return 0
+    else
+        echo >&2 "History check FAILED, cluster upgrade to ${TARGET_VERSION} failed, current version is ${version}, exiting" && return 1
+    fi
+}
+
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# This step slim down for one-hop upgrade only. 
+# If we will call it in chanin upgrade in the future, we need add target loop and health_check back. 
+export OC="run_command_oc"
+# Target version oc will be extract in the /tmp/client directory, use it first
+mkdir -p /tmp/client
+export OC_DIR="/tmp/client"
+export PATH=${OC_DIR}:$PATH
+
+export TARGET="${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}"
```

Comment:
To be more accurate, I think after we determinate `TARGET_VERSION`, we should overwrite the `TARGET` var with the real limage, too.

Reply:
updated